frontend/src/api/models/ModelType.ts

```diff
@@ -9,6 +9,4 @@
  *
  * * `completion`: A language model using completion
  */
-export enum ModelType {
-    COMPLETION = 'completion',
-}
+export type ModelType = string;
```

Comment:
I'm trying to get to the bottom of this before merging. I'm not going to block merge on it but it's annoying to me that Pydantic seems to have changed their behavior in this way

Reply:
Ok, so this is at a crossroads between 3 projects:
- FastAPI
- Pydantic
- openapi-typescript-codegen

What is happening is that before v2, FastAPI would actually be implementing the JSON Schema conversion for Pydantic types that it would emit in openapi.yml. The previous schema for this model looked like this:

```json
      "ModelType": {
        "title": "ModelType",
        "enum": [
          "completion"
        ],
        "type": "string",
        "description": "Enum type indicating the type (i.e. domain) of the registered model.\n\nThe currently supported types are:\n\n* `completion`: A language model using completion"
      }
```

Now though, with Pydantic v2 they merge types that only have one value into the special JSON Schema [`const` type](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-const)

```json
      "ModelType": {
        "type": "string",
        "const": "completion",
        "title": "ModelType",
        "description": "Enum type indicating the type (i.e. domain) of the registered model.\n\nThe currently supported types are:\n\n* `completion`: A language model using completion"
      }
```

This is all perfectly well and good, but slightly annoying as it turns out that openapi-typescript-codegen plugin doesn't seem to understand `const` fields. I've filed a PR to add that though so hopefully they accept

https://github.com/ferdikoomen/openapi-typescript-codegen/pull/1629